CMakeLists.txt

```diff
@@ -1,27 +1,40 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.0)
 project(double-conversion)
 
-include(GNUInstallDirs)
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+set(CMAKE_INCLUDE_CURRENT_DIR_IN_INTERFACE ON)
+
+set(headers
+    double-conversion/bignum.h
+    double-conversion/cached-powers.h
+    double-conversion/diy-fp.h
+    double-conversion/double-conversion.h
+    double-conversion/fast-dtoa.h
+    double-conversion/fixed-dtoa.h
+    double-conversion/ieee.h
+    double-conversion/strtod.h
+    double-conversion/utils.h)
+
+add_library(double-conversion
+            double-conversion/bignum.cc
+            double-conversion/bignum-dtoa.cc
+            double-conversion/cached-powers.cc
+            double-conversion/diy-fp.cc
+            double-conversion/double-conversion.cc
+            double-conversion/fast-dtoa.cc
+            double-conversion/fixed-dtoa.cc
+            double-conversion/strtod.cc
+            ${headers})
 
 # pick a version #
 set(double-conversion_VERSION 2.0.1)
+set_property(TARGET double-conversion PROPERTY VERSION ${double-conversion_VERSION})
 set(double-conversion_SOVERSION_MAJOR 1)
```

Comment:
Version is 2.0.1 but soversion is 1.0.0?

Reply:
Lol I really have no idea what their idea is for versioning. Another question comes to mind, why is version 2.0.1 and not 3.0.0? Really bizarre version scheme here. Don't want to mess with it but IDK.